### PR TITLE
kv (ticdc): remove unused code, simplified some names and added comments to improved the clarity of the code

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: ref #10879 

### What is changed and how it works?

Simplified some names, relocated certain functions, added comments, to improved the clarity of the code.

#### Tests 

 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/kv/main_test.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ func TestMain(m *testing.M) {
 	opts := []goleak.Option{
 		goleak.IgnoreTopFunction("github.com/pingcap/tiflow/pkg/workerpool.(*worker).run"),
 		goleak.IgnoreTopFunction("sync.runtime_Semacquire"),
+		goleak.IgnoreTopFunction("github.com/pingcap/goleveldb/leveldb.(*DB).mpoolDrain"),
```
Comment (CharlesCheung96) [inline cdc/kv/main_test.go:27]: Can we modify the test to avoid leaking goroutine?

File: cdc/kv/shared_client.go [hunk 2]
```diff
@@ -172,20 +172,20 @@ type SharedClient struct {
 	// The tasks will be handled in `handleRangeTask` goroutine.
 	rangeTaskCh *chann.DrainableChann[rangeTask]
 	// regionCh is used to cache region tasks after ranges locked.
```
Comment (CharlesCheung96) [inline cdc/kv/shared_client.go:174]:
```suggestion
	// regionCh is used to cache region tasks have been locked in rangeLock.
```

File: cdc/kv/shared_client.go [hunk 3]
```diff
@@ -172,20 +172,20 @@ type SharedClient struct {
...
-	// The region will be handled in `handleRegionRequest` goroutine.
+	// The region will be handled in `handleRegions` goroutine.
 	regionCh *chann.DrainableChann[regionInfo]
-	// resolveLockCh is used to retrieve resolve lock tasks.
-	resolveLockCh *chann.DrainableChann[resolveLockTask]
-	errCh         *chann.DrainableChann[regionErrorInfo]
+	// resolveLockTaskCh is used to retrieve resolve lock tasks.
```
Comment (CharlesCheung96) [inline cdc/kv/shared_client.go:177]:
```suggestion

	// resolveLockTaskCh is used to retrieve resolve lock tasks.
```

File: cdc/kv/shared_client.go [hunk 4]
```diff
@@ -172,20 +172,20 @@ type SharedClient struct {
...
 
 type resolveLockTask struct {
-	regionID   uint64
-	maxVersion uint64
-	state      *regionlock.LockedRange
-	enter      time.Time
+	regionID uint64
+	targetTs uint64
+	state    *regionlock.LockedRangeState
+	enter    time.Time
```
Comment (CharlesCheung96) [inline cdc/kv/shared_client.go:188]:
```suggestion
	create    time.Time
```

File: cdc/kv/shared_client.go [hunk 5]
```diff
@@ -762,7 +765,7 @@ func (s *SharedClient) handleResolveLockTasks(ctx context.Context) error {
 		}
 	}
 
-	doResolve := func(regionID uint64, state *regionlock.LockedRange, maxVersion uint64) {
+	doResolve := func(regionID uint64, state *regionlock.LockedRangeState, maxVersion uint64) {
```
Comment (CharlesCheung96) [inline cdc/kv/shared_client.go:768]:
```suggestion
	doResolve := func(regionID uint64, state *regionlock.LockedRangeState, targetTs uint64) {
```

File: cdc/kv/shared_client.go [hunk 6]
```diff
@@ -871,20 +874,19 @@ func (s *SharedClient) newSubscribedTable(
 		eventCh:        eventCh,
 	}
 
-	rt.postUpdateRegionResolvedTs = func(regionID, _ uint64, state *regionlock.LockedRange, _ tablepb.Span) {
-		maxVersion := rt.staleLocksVersion.Load()
-		if state.ResolvedTs.Load() <= maxVersion && state.Initialzied.Load() {
+	rt.tryResolveLock = func(regionID uint64, state *regionlock.LockedRangeState) {
+		targetTs := rt.staleLocksTargetTs.Load()
+		if state.ResolvedTs.Load() < targetTs && state.Initialzied.Load() {
```
[the remark this answers is not in the document]
Comment (asddongmen) [reply cdc/kv/shared_client.go:879]: No, `targetTs` is determined by adding a constant value to pd's `currentTs`. The term 'target' indicates our aim to advance the `resolvedTs` to the `targetTs`. Therefore, if the `resolvedTs` is already larger than it, there's no need to send a resolve lock request.

File: cdc/kv/main_test.go (hunk 1 above)
Comment (asddongmen) [reply cdc/kv/main_test.go:27]: No, I merely followed the solution in the TiDB repository, which also had this problem.

**Review by hicqu — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**